### PR TITLE
Add a "User agents that do not support CSS" conformance class.

### LINKED ISSUE #360: Wide Review Comment 2017: Term "WebVTT cue settings list" defined twice
The term `WebVTT cue settings list` is ambiguous because it is defined twice and with different grammars.

## Definition in different chapters

> ### Chapter 4.1

> A WebVTT cue settings list consist of a sequence of zero or more WebVTT cue settings in any order, separated from each other by one or more U+0020 SPACE characters or U+0009 CHARACTER TABULATION (tab) characters. Each setting consists of the following components, in the order given:

> 1. A WebVTT cue setting name. 
> 2. An optional U+003A COLON (colon) character. 
> 3. An optional WebVTT cue setting value. 

> A WebVTT cue setting name and a WebVTT cue setting value each consist of any sequence of one or more characters other than U+000A LINE FEED (LF) characters and - U+000D CARRIAGE RETURN (CR) characters except that the entire resulting string must not contain the substring "-->" (U+002D HYPHEN-MINUS, U+002D HYPHEN-MINUS, U+003E GREATER-THAN SIGN).

**AND**

>### Chapter 4.4

>A WebVTT cue settings list consists of zero or more of the following settings. Each setting must not be included more than once per WebVTT cue settings list.
>
>* A WebVTT vertical text cue setting. 
>* A WebVTT line cue setting. 
>* A WebVTT position cue setting. 
>* A WebVTT size cue setting. 
>* A WebVTT alignment cue setting. 
>* A WebVTT region cue setting. 

## Comment
Although the definition of Chapter 4.4 is compliant with the definition in Chapter 4.1 (it is a specialization of the common syntax), the term `WebVTT cue settings list` can only defined once. Otherwise it is not clear to which the term refers, e.g. in Chapter 4.1:

> A WebVTT cue block consists of the following components, in the given order:
>1. ... 
>2. ...
>3. Optionally, one or more U+0020 SPACE characters or U+0009 CHARACTER TABULATION (tab) characters followed by **a WebVTT cue settings list**.

### LINKED ISSUE #361: Wide Review Comment 2017: Conformance Classes not complete
The following conformance classes are used text but not defined in Section 2.1:

* User agents that don’t support CSS (e.g. Section 7.)
* [...] WebVTT parser [that] requires that the user agent collect WebVTT region settings [..]  

For the later it is implied that a user agent is not always required to collect WebVTT region settings, in which case there will be no region support.

If the first conformance class is added to Section 2.1 this may also require a discussion about the consequences. A user agent that does not support CSS, also does not support text color (see also #279 and #365).

### PR DESCRIPTION
**Pull request by silviapfeiffer**
This actually introduces a definition for "WebVTT cue setting" and
clarifies how they provide information to position and align the cue
box and cue text.

Closes #361

### CODE REVIEW

**Comment by silviapfeiffer** [comment]
ping @tairt for review

**Comment by tairt** [comment]
From my side a merge of this PR needs to wait for a solution of #365. 

See also https://github.com/w3c/webvtt/pull/395#issuecomment-341857127 and https://github.com/w3c/webvtt/issues/361#issuecomment-341900629.

**Comment by silviapfeiffer** [comment]
This should only address #361  and no other issue. Are you ok that this patch addresses #361 ?

**Comment by silviapfeiffer** [comment]
@dwsinger I need a reviewer on this patch

**Comment by silviapfeiffer** [comment]
I'm going to replace /requires to/ with /says to/.

I'm proposing to introduce 3 conformance classes as per #410 - patch upcoming.

File: index.bs [hunk 1]
```diff
@@ -645,7 +645,20 @@ requirements.</p>
...
- <dd><p>All processing requirements in this specification apply, except those in [[#api]].</p></dd>
+ <dd><p>All processing requirements in this specification apply, except those in
+ [[#dom-construction-rules]] and [[#api]].</p></dd>
+
+ <dt><dfn>User agents that do not support CSS</dfn></dt>
+ <dd><p>All processing requirements in this specification apply, except parts of [[#parsing]] that
+ relate to stylesheets and CSS, and all of [[#rendering]] and [[#css-extensions]]. The user agent
+ must instead only render the text inside <a>WebVTT cue text</a> in an appropriate manner. Any other
+ styling and positioning instructions are optional.</p> </dd>
+
```
Comment (tairt) [inline index.bs]: Unless I misunderstood something, we seem to go in circles.

When it says for "User agents that do not support CSS" 

> The user agent must instead only render the text inside <a>WebVTT cue text</a> in an appropriate manner. Any other styling and positioning instructions are optional.

...it seems again also exlcude the required support for color. It even goes further and makes positioning  (which is not defined by CSS extensions but by Cue settings) optional.

**Comment by dwsinger** [comment]
Obviously not clear enough.  Non-CSS UAs are not required to parse and implement STYLE blocks, i.e. new CSS. They must nonetheless respect markup in cues, and the layout model (both of which are CSS-derived).

**Comment by dwsinger** [comment]
I think supporting pre-defined colors is unproblematic, and what we agreed with Andreas...

**Comment by silviapfeiffer** [comment]
It's ok, I only just connected the dots how that referred back to the colors. I'll propose a change next week.

**Comment by tairt** [comment]
@silviapfeiffer Thanks!

**Comment by dwsinger** [comment]
Thanks

I believe that this means that a non-CSS UA is not able to or required to parse STYLE blocks or otherwise allow for the definition of new styles, but is required to follow the built-in layout and styling rules (which can be implemented by, and match the semantics of, CSS).

thanks

> On Jan 5, 2018, at 6:40 , Silvia Pfeiffer <notifications@github.com> wrote:
> 
> I've explicitly stated now that the color classes must be supported by every conformance class, with or without CSS. @tairt could you re-review? Thanks!
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 

David Singer
Manager, Software Standards, Apple Inc.

**Comment by silviapfeiffer** [comment]
Awesome, thanks!

**Comment by dwsinger** [comment]
thank you both for your patience and persistence also

**Pull request merged**